lib/auth/oidc.go

```diff
@@ -455,8 +455,10 @@ func (a *Server) validateOIDCAuthCallback(ctx context.Context, diagCtx *ssoDiagC
 	diagCtx.info.OIDCClaims = types.OIDCClaims(claims)
 
 	log.Debugf("OIDC claims: %v.", claims)
-	if err := checkEmailVerifiedClaim(claims); err != nil {
-		return nil, trace.Wrap(err, "OIDC provider did not verify email.")
+	if !connector.GetAllowUnverifiedEmail() {
+		if err := checkEmailVerifiedClaim(claims); err != nil {
+			return nil, trace.Wrap(err, "OIDC provider did not verify email.")
+		}
 	}
```

Comment:
Do you think it'd be worthwhile to add a test case for that? It's a one small check that might go unnoticed during a refactor for example.

Reply:
I'd agree it would if we had a good way to test this. I couldn't figure out a could way to unit-test this whole verification function and I am mostly-unaware of any tests for doing so.